app/Libraries/Payments/PaymentProcessor.php

```diff
@@ -216,7 +216,7 @@ public function cancel()
         $this->sandboxAssertion();
 
         if (!$this->validateTransaction()) {
-            $this->throwValidationFailed(new PaymentProcessorException($this->validationErrors()));
+            $this->throwValidationFailed(new PaymentProcessorException($this->getOrder(), $this->validationErrors()));
```

Comment:
not related to this pr but it looks like there should be an `assertValidTransaction` which validates transaction and throw if not valid